protocols/relay/src/v2/relay.rs

```diff
@@ -213,8 +213,10 @@ impl NetworkBehaviour for Relay {
         connection: &ConnectionId,
         _: &ConnectedPoint,
         _handler: Either<handler::Handler, DummyProtocolsHandler>,
+        _remaining_established: usize,
     ) {
         if let Some(connections) = self.reservations.get_mut(peer) {
+            // TODO: it looks like `peer` is never removed from `self.reservations`?
```

Comment:
Good catch! Mind taking a look at https://github.com/libp2p/rust-libp2p/pull/2464 @divagant-martian?

Reply:
done. I completely forgot about this finding after writing this, ty for addressing it

Reply:
This was perfect. Big fan of `TODO` comments up until code is merged to `master`. Thanks.

With https://github.com/libp2p/rust-libp2p/pull/2464 merged, I think this can now be removed.

Reply:
nice. removed